tensorboard/plugins/hparams/tf_hparams_backend/tf-hparams-backend.html

```diff
@@ -29,37 +29,50 @@
   (function (hparams) {
     class Backend {
       // Constructs a backend that uses the given tf_backend.requestManager to
-      // send "GET" requests. The apiUrl parameter should denote the base
-      // url to use.
-      constructor(apiUrl, requestManager) {
+      // send requests. The apiUrl parameter should denote the base
+      // url to use. If useHttpGet is true uses HTTP GET to send a request
+      // otherwise uses HTTP POST. See tensorboard/plugins/hparams/http_api.md
+      // for details on how the requests and responses are encoded in each
+      // scheme (GET or POST).
+      constructor(apiUrl, requestManager, useHttpGet=true) {
         this._apiUrl = apiUrl;
         this._requestManager = requestManager;
+        this._useHttpGet = useHttpGet;
       }
 
       // In the API methods below, 'request' should be a JSON translated request
       // protocol buffer and the response is a JSON translated response protocol
       // buffer. See api.proto for the details.
 
       getExperiment(experimentRequest) {
-        return this._requestManager.request(
-            this._buildUrl('experiment', experimentRequest));
+        return this._sendRequest('experiment', experimentRequest);
       }
 
       listSessionGroups(listSessionGroupsRequest) {
-        return this._requestManager.request(
-            this._buildUrl('session_groups', listSessionGroupsRequest));
+        return this._sendRequest('session_groups', listSessionGroupsRequest);
       }
 
       listMetricEvals(listMetricEvalsRequest) {
-        return this._requestManager.request(
-            this._buildUrl('metric_evals', listMetricEvalsRequest));
+        return this._sendRequest('metric_evals', listMetricEvalsRequest);
       }
 
       // ---- Private methods below -------------------------------------------
-      _buildUrl(methodName, request_proto) {
-        const encodedRequest = encodeURIComponent(
-            JSON.stringify(request_proto));
-        return this._apiUrl + '/' + methodName + '?request=' + encodedRequest;
+      _sendRequest(methodName, request_proto) {
+        if (this._useHttpGet) {
+          const encodedRequest = encodeURIComponent(
+              JSON.stringify(request_proto));
+          const url = this._apiUrl + '/' + methodName +
+                '?request=' + encodedRequest;
+          return this._requestManager.request(url);
+        }
+        /* Use POST */
+        const requestOptions = new tf_backend.RequestOptions();
+        requestOptions.withCredentials = true;
+        requestOptions.methodType = 'POST';
+        requestOptions.contentType = 'text/plain';
+        requestOptions.body = JSON.stringify(request_proto);
+        const url = this._apiUrl + '/' + methodName;
+        return this._requestManager.requestWithOptions(url, requestOptions);
```

Comment:
You are invoking a private method here. Please consider making this a public API (which I think is reasonable).

Reply:
Yes that was my intention. The private was a mistake.